src/requirements/courses-json-generator.ts

```diff
@@ -1,17 +1,211 @@
+// Disabled because we want to log progress.
+/* eslint-disable no-await-in-loop */
+// Disabled because we want the inefficiency to rate-limit ourselves.
+/* eslint-disable no-console */
+import fetch from 'node-fetch';
 import { writeFileSync } from 'fs';
-import { courseCodeToDataAndOfferedMap } from './filtered-course-data';
 import { Course } from './types';
 
-const fullCourseJson = (() => {
-  const json: Record<number, Course & { roster: string }[]> = {};
+const PREFIX = 'https://classes.cornell.edu/api/2.0';
+
+const wait = (time: number) => new Promise<void>(resolve => setTimeout(() => resolve(), time));
+
+const getSemesters = async (): Promise<readonly string[]> => {
```

Comment:
Can you document this code (like each `const` or at least in the form of a spec at the top of this script) to show that we're ultimately getting the latest offering of a course and then combining them into the full-courses.json?